src/conn/conn_capacity.c

```diff
@@ -34,14 +34,26 @@ __capacity_config(WT_SESSION_IMPL *session, const char *cfg[])
     uint64_t chunkcache, total;
 
     conn = S2C(session);
-    chunkcache = total = 0;
+    cap = &conn->capacity;
 
+    chunkcache = total = 0;
     WT_RET(__wt_config_gets(session, cfg, "io_capacity.total", &cval));
-    if (cval.val != 0) {
-        if (cval.val < WT_THROTTLE_MIN)
-            WT_RET_MSG(session, EINVAL, "total I/O capacity value %" PRId64 " below minimum %d",
-              cval.val, WT_THROTTLE_MIN);
-        total = (uint64_t)cval.val;
+    if (cval.val != 0 && cval.val < WT_THROTTLE_MIN)
+        WT_RET_MSG(session, EINVAL, "total I/O capacity value %" PRId64 " below minimum %d",
+          cval.val, WT_THROTTLE_MIN);
+    cap->total = total = (uint64_t)cval.val;
+
+    WT_RET(__wt_config_gets(session, cfg, "io_capacity.fsync_maximum_wait_period", &cval));
+    if (cval.val != 0 && cval.val > WT_FSYNC_BACKGROUND_MAX_PERIOD_SEC)
+        WT_RET_MSG(session, EINVAL,
+          "fsync_maximum_wait_period I/O capacity value %" PRId64 " exceeds maximum %d", cval.val,
+          WT_FSYNC_BACKGROUND_MAX_PERIOD_SEC);
+    cap->fsync_maximum_wait_period = (uint64_t)cval.val;
```

Comment:
There is already a check for over 60, it is enforced because of the new code in api_data.py, where `max='60'`.  So I would change this to merely:
```
    if (cval.val != 0)
       cap->fsync_maximum_wait_period = (uint64_t)cval.val;
```

I wanted to point out though, that the construct in the PR at line 47 can be unsafe.  That is, assume there was not the upper and lower limit check in api_data.py.  Then the caller could have this in the config string:
```
    io_capacity=(fsync_maximum_wait_period=-10)
```
cval.val is a signed value, so the `if` clause would _not_ be taken, no error issued and the line `cap->fsync_maximum_wait_period = (uint64_t)cval.val;` would get executed (instead of issuing the error).  I would avoid mistakes like this by using this idiom:
```
    if (cval.val != 0) {
        ...any other checks...
        cap->.... = (uint64_t)cval.val;
   }
```
It would be good to use this idiom whenever checking config strings.